src/LibVulkan.jl

```diff
@@ -2,29 +2,25 @@ module LibVulkan
 
 import Libdl
 
-@static if Sys.iswindows()
-    const libvulkan = "vulkan-1.dll" 
-elseif Sys.isapple()
-    const libvulkan = "libvulkan.dylib"
-elseif Sys.islinux()
-    const libvulkan = "libvulkan.so.1"
-else
-    const libvulkan = "libvulkan"
+const libnames = ["vulkan-1", "vulkan", "libvulkan", "libvulkan.so.1", "libvk_swiftshader"]
+
+const libextrapaths = []
+
+foreach([get(ENV, "JULIA_VULKAN_LIBNAME", ""), get(ENV, "JULIA_VULKAN_SDK_LIBNAME", "")]) do libname
+    !isempty(libname) && push!(libnames, libname)
 end
 
-libvulkan_handle = C_NULL
```

Comment:
Indeed, `const libvulkan_handle = Ref{Ptr{Cvoid}}(0)` is better.

Reply:
But why did we have this variable? It is never used outside being assigned from `__init__`. Am I missing something?

Reply:
IIRC, if we `dlopen` the library here, those "libname"-ccalls will immediately find the path.

Reply:
From [this comment](https://github.com/JuliaLang/julia/issues/7004#issuecomment-44680170) it looks like there is some kind of caching behavior after you `dlopen` yes. But `find_library` already calls `dlopen`, so is it really needed? I haven't found other packages that `dlopen` their library explicitly like this.

Reply:
Well, you can find it in every _jll packages... 

https://github.com/JuliaBinaryWrappers/CImGui_jll.jl/blob/3d869e10f5c4e4d4f9fc961f7beae209480f3033/src/wrappers/x86_64-apple-darwin14.jl#L57-L78

Reply:
Oh. My bad. Thanks

Reply:
> But find_library already calls dlopen, so is it really needed? I haven't found other packages that dlopen their library explicitly like this.

I agreed on this point. I didn't know the implementation details of `find_library`.

Reply:
> I didn't know the implementation details of `find_library`.
It `dlclose`s the handle afterwards. If I understood it right, it shouldn't matter, but I am now wondering why the _jll_ packages don't close it afterwards.